webapp-django/crashstats/crashstats/views.py

```diff
@@ -817,24 +893,293 @@ def append_row_blob(blob, labels):
+        date_range_types=date_range_types,
+        hang_types=hang_types,
+    )
+    if form.is_valid():
+        params = form.cleaned_data
+        params['product'] = params.pop('p')
+        params['versions'] = sorted(list(set(params.pop('v'))), reverse=True)
         try:
-            blob = data['totals'][product_version]
+            params['platforms'] = params.pop('os')
         except KeyError:
-            continue
-        append_row_blob(blob, totals_labels)
-    writer.writerow(row)
-    return response
+            params['platforms'] = None
+    else:
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    if len(params['versions']) > 0:
+        context['version'] = params['versions'][0]
+
+    context['product'] = params['product']
+
+    if not params['versions']:
+        # need to pick the default featured ones
+        params['versions'] = [
+            version['version']
+            for version in context['currentversions']
+            if version['product'] == params['product'] and version['featured']
+        ]
+
+    context['available_versions'] = []
+    now = datetime.datetime.utcnow().date()
+    for version in context['currentversions']:
+        start_date = isodate.parse_date(version['start_date'])
+        end_date = isodate.parse_date(version['end_date'])
+        if (
+            params['product'] == version['product'] and
+            start_date <= now and
+            end_date >= now
+        ):
+            context['available_versions'].append(version['version'])
+
+    if not params.get('platforms'):
+        params['platforms'] = [
+            x['name'] for x in platforms if x.get('display')
+        ]
+    context['platforms'] = params.get('platforms')
+
+    end_date = params.get('date_end') or datetime.datetime.utcnow()
+    if isinstance(end_date, datetime.datetime):
+        end_date = end_date.date()
+    start_date = (params.get('date_start') or
+                  end_date - datetime.timedelta(weeks=2))
+    if isinstance(start_date, datetime.datetime):
+        start_date = start_date.date()
+
+    context['start_date'] = start_date.strftime('%Y-%m-%d')
+    context['end_date'] = end_date.strftime('%Y-%m-%d')
+
+    context['duration'] = abs((start_date - end_date).days)
+    context['dates'] = utils.daterange(start_date, end_date)
+
+    context['hang_type'] = params.get('hang_type') or 'any'
+
+    default = request.session.get('date_range_type', 'report')
+    context['date_range_type'] = params.get('date_range_type') or default
+
+    # if params.get('hang_type') == 'any':
+    #     hang_type = None
+    # else:
+    #     hang_type = params.get('hang_type')
+
+    api = SuperSearchUnredacted()
+    # XXX HACK! Because of a bug in SuperSearch we can't search for
+    # '41.0b' so we have to change that to '41.0b1' :(
+    # See https://bugzilla.mozilla.org/show_bug.cgi?id=1197251
+    if '41.0b' in params['versions']:
+        params['versions'].insert(
```

Comment:
Order matters for the 4 version select boxes. I want it to be be 43, 42, 41.0b1, 40. Not 43, 42, 40, 41.0b1